source/agora/consensus/data/Block.d

```diff
@@ -601,32 +598,32 @@ version (unittest)
         auto revealed = key_pairs.enumerate.filter!(en => !missing_validators.canFind(en.index)).map!(en => en.value).array;
         Hash[] pre_images = wellKnownPreimages(prev_block.header.height + 1, revealed);
         assert(revealed.length == key_pairs.length - missing_validators.length);
-        Hash random_seed = createRandomSeed(pre_images);
-
-        // the time_offset passed to makeNewBlock should really be
-        // prev_block.header.time_offset + ConsensusParams.BlockInterval instead of
-        // prev_block.header.time_offset + 1
-        // however many tests calling makeNewTestBlock have no access to ConsensusParams
-        auto block = makeNewBlock(prev_block, txs,
-                time_offset ? time_offset : prev_block.header.time_offset + 1,
-                random_seed, key_pairs.length, enrollments, missing_validators);
-        auto validators = BitField!ubyte(key_pairs.length);
-        Signature[] sigs;
-        ulong offset = 0;
-        key_pairs.enumerate.each!((i, k)
-        {
-            if (missing_validators.canFind(i))
-            {
-                log.trace("Skip signing with validator idx {} as it is in missing_validators", i);
-            }
-            else
-            {
-                validators[i] = true;
-                sigs ~= block.header.createBlockSignature(k.secret, offset);
-            }
-        });
         try
         {
+            Hash random_seed = createRandomSeed(pre_images);
+
+            // the time_offset passed to makeNewBlock should really be
+            // prev_block.header.time_offset + ConsensusParams.BlockInterval instead of
+            // prev_block.header.time_offset + 1
+            // however many tests calling makeNewTestBlock have no access to ConsensusParams
+            auto block = makeNewBlock(prev_block, txs,
+                    time_offset ? time_offset : prev_block.header.time_offset + 1,
+                    random_seed, key_pairs.length, enrollments, missing_validators);
+            auto validators = BitField!ubyte(key_pairs.length);
+            Signature[] sigs;
+            ulong offset = 0;
+            key_pairs.enumerate.each!((i, k)
+            {
+                if (missing_validators.canFind(i))
+                {
+                    log.trace("Skip signing with validator idx {} as it is in missing_validators", i);
```

Comment:
Since the logger is only used for this single statement, which is a test utility, I'd say we should remove it.

Reply:
ok will do.